web/libs/editor/src/components/InteractiveOverlays/CommentsOverlay.tsx

```diff
@@ -0,0 +1,193 @@
+    <g
+      className={itemStyles.join(" ")}
+      style={positionStyle}
+      onMouseEnter={onHover}
+      onMouseLeave={onUnHover}
+      onClick={comment.scrollIntoView}
+    >
+      <CommentIcon />
+    </g>
+  );
+});
+
+type CommentsOverlayProps = {
+  commentStore: MSTCommentStore;
+  annotation: MSTAnnotation;
+};
+const CommentsOverlayInner: React.FC<CommentsOverlayProps> = observer(({ annotation, commentStore }) => {
+  const { overlayComments } = commentStore || {};
+  const rootRef = useRef<SVGSVGElement>();
+  const [uniqKey, forceUpdate] = useState<any>(guidGenerator());
+
+  const mounted = useMounted();
+
+  const loadComments = async () => {
+    await commentStore.listComments({ mounted, suppressClearComments: commentStore.isRelevantList });
+  };
+
+  useEffect(() => {
+    loadComments();
+    // id is internal id,
+    // always different for different annotations, even empty ones;
+    // remain the same when user submit draft, so no unneeded calls.
+  }, [commentStore.annotation?.id]);
+
+  const resizeObserver: ResizeObserver = useMemo(() => {
+    let requestId: number;
+
+    return new ResizeObserver((entities) => {
+      cancelAnimationFrame(requestId);
+      requestId = requestAnimationFrame(() => {
+        forceUpdate(guidGenerator());
+      });
+    });
+  }, []);
+
+  useEffect(() => {
+    return () => {
+      resizeObserver?.disconnect();
+    };
+  }, []);
+  const setRef = useCallback((ref: SVGSVGElement | null) => {
+    const refOfFixedType = ref || undefined;
+    if (ref) {
+      resizeObserver?.observe(ref);
+      resizeObserver?.observe(document.body);
+    } else if (rootRef.current) {
+      resizeObserver?.unobserve(rootRef.current);
+      resizeObserver?.unobserve(document.body);
+    }
+    rootRef.current = refOfFixedType;
+  }, []);
+
+  if (!overlayComments) return null;
+
+  const containerStyles = [styles.container];
+  if (commentStore?.isHighlighting) {
+    containerStyles.push(styles._highlighting);
+  }
+
+  return (
+    // biome-ignore lint/a11y/noSvgWithoutTitle: It's not just an icon or a figure; it's an entire interactive layer.
+    <svg className={containerStyles.join(" ")} ref={setRef} xmlns="http://www.w3.org/2000/svg">
+      {overlayComments.map((comment: MSTComment) => {
+        const { id } = comment;
+        return <CommentItem key={[id, uniqKey].join("-")} comment={comment} rootRef={rootRef} />;
+      })}
+    </svg>
+  );
+});
+
+type WhenTagsReadyProps = {
+  tags: Map<string, MSTTag>;
+  children: React.ReactNode;
+};
+const WhenTagsReady: React.FC<WhenTagsReadyProps> = memo(
+  observer(({ tags, children }) => {
+    if (
+      !Array.from(tags.values()).every((tag) => {
+        if (!isAlive(tag)) return false;
+        // Unfortunately, for some reason image do not use `isReady` in the way it was planed to do, so we need to check it separately
```

Comment:
interesting, if you haven't yet - please create a ticket to investigate this so it doesn't fall through the cracks

Reply:
and as I understand it works weird if you have FF for image preload disabled